src/containers/shared/components/DomainLink.tsx

```diff
@@ -14,7 +14,7 @@ const DomainLink = (props: Props) => {
       className={classnames(`domain`, className)}
       rel="noopener noreferrer"
       target="_blank"
-      href={`https://${decode ? decodeHex(domain) : domain}`}
+      href={decode ? domain : `https://${domain}`}
```

Comment:
If the decode property is set then we need to decode the domain which is always in hexadecimal.  The issue is that the decoded value sometimes starts with the protocol like `https://`.  If that is case we need not add http `https://`

Reply:
The reason why the tests are failing is that after your change when you specify decode it is outputting the domain in hex.

You also need to add a new test that when the domain contains the protocol it needs to not double the protocol.

Reply:
Yes. Thanks for the follow up. I'll revise it.

Reply:
I added the test as such:
```typescript
  it('handles domain link with domain containing protocol', () => {
    const wrapper = mount(<DomainLink domain="https://example.com" />)
    expect(wrapper.find('a').props().className).toEqual('domain')
    expect(wrapper.find('a').text()).toEqual('https://example.com')
    expect(wrapper.find('a').props().href).toEqual('https://example.com')
    wrapper.unmount()
  })
```
Would the above be viable?

Reply:
You actually need to test the input as coming in encoded as HEX.

```
const wrapper = mount(<DomainLink domain=" 68747470733A2F2F6578616D706C652E636F6D" />)
```

Reply:
I've changed it as suggested.
Take a look:
```typescript
  it('handles domain link with domain provided in HEX-encoded format', () => {
    const wrapper = mount(
      <DomainLink domain="68747470733A2F2F6578616D706C652E636F6D" decode />,
    )
    // The HEX-encoded input should be decoded to 'example.com'
    expect(wrapper.find('a').props().className).toEqual('domain')
    expect(wrapper.find('a').text()).toEqual('https://example.com')
    expect(wrapper.find('a').props().href).toEqual('https://example.com')
    wrapper.unmount()
  })

```